src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -1385,14 +1369,18 @@ static void buildIndexes(
     if (!indexes->use_skip_indexes)
         return;
 
-    const SelectQueryInfo * info = &query_info;
     std::optional<SelectQueryInfo> info_copy;
-    if (settings.allow_experimental_analyzer)
+    auto get_query_info = [&]() -> const SelectQueryInfo &
```

Comment:
This is not good, if we call `get_query_info` multiple times we unnecessarily create multiple copy of `query_info`.